tests/onnx/quantization/regression_test/test_ptq_regression.py

```diff
@@ -0,0 +1,130 @@
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import pytest
+
+import copy
+from pathlib import Path
+import nncf
+import numpy as np
+import onnx
+from onnx import version_converter
+import openvino.runtime as ov
+import torch
+from fastdownload import FastDownload
+from sklearn.metrics import accuracy_score
+from torchvision import datasets, transforms
+from tqdm import tqdm
+
+MODELS = [
+    ('https://github.com/onnx/models/raw/main/vision/classification/mobilenet/model/mobilenetv2-12.onnx',
+     'mobilenetv2-12', 0.7875159235668789),
+    ('https://github.com/onnx/models/raw/main/vision/classification/resnet/model/resnet50-v1-7.onnx',
+     'resnet50-v1-7', 0.8119745222929936),
+    (
+    'https://github.com/onnx/models/raw/main/vision/classification/efficientnet-lite4/model/efficientnet-lite4-11.onnx',
+    'efficientnet-lite4-11', 0.8010191082802548)
+]
+
+DATASET_URL = 'https://s3.amazonaws.com/fast-ai-imageclas/imagenette2-320.tgz'
+DATASET_PATH = '~/.cache/nncf/datasets'
+
+
+def download_dataset() -> Path:
+    downloader = FastDownload(base=DATASET_PATH,
+                              archive='downloaded',
+                              data='extracted')
+    return downloader.get(DATASET_URL)
+
+
+def download_model(model_url, tmp_path) -> Path:
+    downloader = FastDownload(base=tmp_path)
+    return downloader.download(model_url)
+
+
+def validate(quantized_model_path: Path, data_loader: torch.utils.data.DataLoader) -> float:
+    from_imagenet_to_imageneetee = {
+        0: 0,  # tench
+        217: 1,  # English springer
+        482: 2,  # cassette player
+        491: 3,  # chain saw
+        497: 4,  # church
+        566: 5,  # French horn
+        569: 6,  # garbage truck
+        571: 7,  # gas pump
+        574: 8,  # golf ball
+        701: 9  # parachute
+    }
+    compiled_model = ov.compile_model(quantized_model_path)
+    infer_queue = ov.AsyncInferQueue(compiled_model)
+
+    predictions = [0] * len(data_loader)
+    references = [-1] * len(data_loader)
+
+    def res_callback(infer_request: ov.InferRequest, userdata) -> None:
+        pred = infer_request.get_output_tensor().data
+        pred_class = np.argmax(pred, axis=1)
+        if pred_class.item() in from_imagenet_to_imageneetee:
+            pred_class[0] = from_imagenet_to_imageneetee[pred_class.item()]
+        predictions[userdata] = [pred_class]
+
+    infer_queue.set_callback(res_callback)
+
+    for i, (images, target) in tqdm(enumerate(data_loader)):
+        # W/A for memory leaks when using torch DataLoader and OpenVINO
+        image_copies = copy.deepcopy(images.numpy())
```

Comment:
How does use `deepcopy` for every image batch reduces memory consumption?

Reply:
Removed